contracts/AMM/BalancerAMM.sol

```diff
@@ -128,14 +137,35 @@ contract BalancerAMM {
         uint256 tokenAmount,
         uint256 amountOutMin
     ) external returns (uint256 sweepAmount) {
+        bool lowerPriceInPool = true;
+
+        if (address(marketMaker) != address(0)) {
+            uint256 buyPrice = marketMaker.getBuyPrice();
+            if (buyPrice < getPrice()) {
+                lowerPriceInPool = false;
+                uint256 usdxInUsd = ChainlinkLibrary.convertTokenToUsd(tokenAmount, base.decimals(), oracleBase);
+                sweepAmount = sweep.convertToSWEEP(usdxInUsd);
+                uint256 slippage = ((sweepAmount/amountOutMin)-1) * USD_DECIMALS;
```

Comment:
I think USD_DECIMALS should be base.decimals().
Also it would be better to do
`sweepAmount * base.decimals() / amountOutMint - base.decimals()`
to avoid losing precision.